setup.py

```diff
@@ -69,7 +69,6 @@
     license='MIT',
     scripts=[],
     data_files=[],
-    setup_requires=["oldest-supported-numpy", "cython>=0.23"],
```

Comment:
`setup_requires` is deprecated and indeed trying to use it actually breaks a NumPy 2.0 build, so this needs to get removed. The `build-system` information in `pyproject.toml` supersedes it.

c.f. https://learn.scientific-python.org/development/guides/packaging-classic/#pep-517518-support-high-priority